src/datadog/grok/filters/keyvalue.rs

```diff
@@ -302,84 +259,50 @@ fn quoted<'a>(
+/// 2. Otherwise, we parse until regex matches
 #[inline]
-fn parse_value<'a>(
-    field_delimiter: &'a str,
-    quotes: &'a [(char, char)],
-    re: &'a Regex,
-) -> impl Fn(&'a str) -> SResult<Value> {
-    move |input| {
-        alt((
-            map(quoted(quotes, field_delimiter), |value| {
-                Value::Bytes(Bytes::copy_from_slice(value.as_bytes()))
-            }),
-            parse_null,
-            parse_boolean,
-            parse_number(field_delimiter),
-            map(match_re_or_empty(re, field_delimiter), |value| {
-                Value::Bytes(Bytes::copy_from_slice(value.as_bytes()))
-            }),
-        ))(input)
-    }
+fn parse_value<'a>(input: &'a str, quotes: &'a [(char, char)]) -> SResult<'a, Value> {
+    alt((
+        parse_null,
+        parse_boolean,
+        parse_number,
+        quoted(quotes).and_then(parse_string),
+        parse_string,
+    ))(input)
 }
 
-fn parse_number(field_delimiter: &str) -> impl Fn(&str) -> SResult<Value> + '_ {
-    move |input| {
-        map(
-            terminated(
-                double,
-                peek(alt((
-                    parse_field_delimiter(field_delimiter),
-                    parse_end_of_input(),
-                ))),
-            ),
-            |v| {
-                if ((v as i64) as f64 - v).abs() == 0.0 {
-                    // can be safely converted to Integer without precision loss
-                    Value::Integer(v as i64)
-                } else {
-                    Value::Float(NotNan::new(v).expect("not a float"))
-                }
-            },
-        )(input)
-        .map_err(|e| match e {
-            // double might return Failure(an unrecoverable error) - make it recoverable
-            nom::Err::Failure(_) => nom::Err::Error((input, nom::error::ErrorKind::Float)),
-            e => e,
-        })
+fn parse_string(input: &str) -> SResult<Value> {
+    map(rest, |s: &str| {
+        Value::Bytes(Bytes::copy_from_slice(s.trim().as_bytes()))
+    })(input)
+}
+
+fn parse_number(input: &str) -> SResult<Value> {
+    let res = map(terminated(double, eof), |v| {
+        if ((v as i64) as f64 - v).abs() == 0.0 {
+            // can be safely converted to Integer without precision loss
+            Value::Integer(v as i64)
+        } else {
+            Value::Float(NotNan::new(v).expect("not a float"))
+        }
+    })(input)
+    .map_err(|e| match e {
+        // double might return Failure(an unrecoverable error) - make it recoverable
+        nom::Err::Failure(_) => nom::Err::Error((input, nom::error::ErrorKind::Float)),
+        e => e,
+    });
+    match res {
+        // check if it is a valid octal number(start with 0) - keep parsed as a decimal though
+        Ok((_, Value::Integer(_)))
+            if input.starts_with('0') && input.contains(|c| c == '8' || c == '9') =>
+        {
+            Err(nom::Err::Error((input, nom::error::ErrorKind::OctDigit)))
+        }
+        res => res,
     }
```

Comment:
Would this make sense to put above in the single branch that creates the `Value::Integer` variant? Also, does this secondary parsing now make this PR a breaking change, as the string `015` would parse as 15 before and 13 after the change?

Reply:
Agree - extracted a function https://github.com/vectordotdev/vrl/pull/1015/commits/b4d67ad379fbb035d8c39c137bc8c894ab7fb1c8.
As to parsing octal numbers - we keep parsing them as decimal(no octal to decimal conversion), which corresponds to Datadog grok implementation.